docs/ruleset_functions/compare_swh_dist_systems_and_components.md

```diff
@@ -0,0 +1,54 @@
+## compare_swh_dist_systems_and_components
+
+Description: This function compares all sub-components of a given SWH Distribution System ID between two models.  The function also accepts   
+
+Inputs:
+- **RMD1**
+- **RMD2**
+- **compare_context_str**
+- **swh_distribution_id**
+- **errors**
+
+Returns:
+- **all_match**: A boolean indicating whether all of the SWH elements match for the given SWH equipment id in models 1 and 2
+
+Function Call:
+
+- **compare_context_pair** - there is no RDS for this function, but it is a function developed for Rule 1-6 that compares two elements  
+- **get_component_by_id**
+
+Data Lookup: None
+
+Logic:
+  - use the function get_swh_components_associated_with_each_swh_distribution_system to get a dictionary of SWH equipment associated with each Distribution System for model 1 (pumps, tanks, SWH use, distribution system, etc): `RMD1_swh_system_and_equip_dict = get_swh_components_associated_with_each_swh_distribution_system(RMD1)`
+  - use the function get_swh_components_associated_with_each_swh_distribution_system to get a dictionary of SWH equipment associated with each Distribution System for model 1 (pumps, tanks, SWH use, distribution system, etc): `RMD2_swh_system_and_equip_dict = get_swh_components_associated_with_each_swh_distribution_system(RMD2)`
+
+  - check if this distribution ID is in model 1: `if RMD1_swh_system_and_equip_dict[swh_distribution_id]:`
+    - get the dictionary of all SWH equipment (pumps, tanks, SWH use, dist system, etc) for the given swh_distribution_id for RMD1: `RMD1_swh_equipment_dict = RMD1_swh_system_and_equip_dict[swh_distribution_id]`
+    - get the actual distribution equipment associated with this ID from model 1: `RMD1_swh_distribution = get_component_by_id(RMD1, swh_distribution_id)`
+  - otherwise, return FALSE: `else: errors << "SWH Distribution System " + swh_distribution_id + " not found in one of the two RMDs"; return FALSE`
+
+  - check if this distribution ID is in model 2: `if RMD2_swh_system_and_equip_dict[swh_distribution_id]:`
+    - get the dictionary of all SWH equipment (pumps, tanks, SWH use, dist system, etc) for the given swh_distribution_id for RMD2: `RMD2_swh_equipment_dict = RMD2_swh_system_and_equip_dict[swh_distribution_id]`
+    - get the actual distribution equipment associated with this ID from model 1: `RMD2_swh_distribution = get_component_by_id(RMD2, swh_distribution_id)`
+  - otherwise, add an error and return FALSE: `else: errors << "SWH Distribution System " + swh_distribution_id + " not found in one of the two RMDs"; return FALSE`
```

Comment:
How is this error message delivered to the reviewer since this function only retyrns a boolean?

Reply:
because errors is a list that was passed into this function, any items added to the list will be accessible by the rule that called that function.

Reply:
I'll have to trust you on that. Not sure my python knowledge is good enough to understand how this can be accessible if the function only returns TRUE or FALSE.

Reply:
recommend writing that the function returns the errors and the boolean because this is psuedo code we don't want any assumptions about the scope of variables to be miscommunicated

Reply:
Karen going to add note to describe that error is returned

Reply:
I left the same comment - return a list makes sense.